addons/account/models/account_move.py

```diff
@@ -1892,6 +1899,18 @@ def _compute_taxes_legal_notes(self):
                 if not is_html_empty(tax.invoice_legal_notes)
             )
 
+    @api.depends('line_ids.payment_date', 'line_ids.reconciled')
```

Comment:
This is a soft conflict with my branch of optional journal entries because I think that we should also consider the pending payments, not only the reconciled ones